### PR TITLE
adds support for verbose response when checking domains

### LINKED ISSUE #189: Return verbose response for green hosting checks
**Is your feature request related to a problem? Please describe.**
Currently, running a greencheck using CO2.js returns either: 

- a boolean (`true`/`false`) when checking a single domain
- an array of domains when checking multiple domains

**Describe the solution you'd like**
It would be nice to have the option to get back a more verbose response from the Greencheck API, including any public evidence that is available for a green provider.

**Describe alternatives you've considered**
A developer could perform this fetch request themselves, but it would be nice to have it available within the library.

**Additional context**
For example, checking "google.com" returns:

https://api.thegreenwebfoundation.org/api/v3/greencheck/google.com

```json
{
  "url": "google.com",
  "hosted_by": "Google Inc.",
  "hosted_by_website": "https://www.google.com",
  "partner": null,
  "green": true,
  "hosted_by_id": 595,
  "modified": "2024-02-02T04:27:28",
  "supporting_documents": [
    {
      "id": 108,
      "title": "Sustainability at Google",
      "link": "https://sustainability.google"
    },
    {
      "id": 139,
      "title": "Independent verification of Google 2020 Reporting",
      "link": "https://s3.nl-ams.scw.cloud/tgwf-web-app-live/uploads/Google_Cloud_-_3degrees_cloud_services_review_statement_final.pdf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=SCWT1WBAW6NZ5SW5GYJ8%2F20240202%2Fnl-ams%2Fs3%2Faws4_request&X-Amz-Date=20240202T054248Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=1976a86d90604c11260b17420839a2c92b1dcdfba49b867dc23436ece5585fbf"
    },
    {
      "id": 140,
      "title": "2021 Environmental Report",
      "link": "https://s3.nl-ams.scw.cloud/tgwf-web-app-live/uploads/google-2021-environmental-report.pdf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=SCWT1WBAW6NZ5SW5GYJ8%2F20240202%2Fnl-ams%2Fs3%2Faws4_request&X-Amz-Date=20240202T054248Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=c97f46d056156a53201743603de3d863688f3c9aedf57e91ee29d94fe7171c9e"
    },
    {
      "id": 141,
      "title": "2022 Environmental Report",
      "link": "https://s3.nl-ams.scw.cloud/tgwf-web-app-live/uploads/google-2022-environmental-report.pdf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=SCWT1WBAW6NZ5SW5GYJ8%2F20240202%2Fnl-ams%2Fs3%2Faws4_request&X-Amz-Date=20240202T054248Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=a7346f2bdd14f7a18ee87a9524c4bb5b4c8fbeb1f6847dbac2439fdab2e71185"
    },
    {
      "id": 142,
      "title": "Independent verification of Google 2021 Reporting",
      "link": "https://s3.nl-ams.scw.cloud/tgwf-web-app-live/uploads/alphabet-fy2021-environmental-indicators-assurance-letter.pdf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=SCWT1WBAW6NZ5SW5GYJ8%2F20240202%2Fnl-ams%2Fs3%2Faws4_request&X-Amz-Date=20240202T054248Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=965f57864647a0313f2e0f75a6c0394b6e8441f61e962b2bebdce22ced830ee1"
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by sfishel18**
- adds a new `checkVerbose` function to both hosting-node.js and hosting-api.js
- for checking a single domain, `checkVerbose` returns the entire payload from the api.thegreenwebfoundation.org endpoint
- for checking multiple domains, `checkVerbose` returns a dictionary of each domain mapped to its response payload

Fixes #189

### CODE REVIEW

**Comment by sfishel18** [comment]
@fershad / @mrchrisadams here's my initial implementation, let me if this looks like a good direction and if so i'll polish it up and add more test cases

**Comment by sfishel18** [comment]
@fershad i think this is ready for an initial look from you. i plan to add additional tests for the new code paths, but want to make sure i'm on the right track.

a couple of specific things i wanted your feedback on:

1. if i understood your comments on the issue, a goal of this is to make the function signatures for the two `check` functions consistent. i *think* it's possible to do that without breaking backwards compatibility, by doing introspection of the argument types. but the "overloaded" function signature is a little confusing, so i'd understand if you'd prefer to just go ahead and make a breaking change to the API.

2. from what i can tell, the local database only contains a list of string domain names, so it doesn't have enough information to return verbose responses. i put in logic to throw an exception if a local database is used in combination with the `verbose` flag set.

File: src/hosting-node.js [hunk 1]
```diff
@@ -51,59 +52,82 @@ async function getBody(url, userAgentIdentifier) {
...
+ *   representing the app, site, or organisation that is making the request.
+ * @returns - A boolean if a string was provided, or an array of booleans if an array of domains was provided.
+ *   if a string was provided for `domain`: a boolean indicating whether the domain is hosted by a green web host if `options.verbose` is false,
+ *     otherwise an object representing the domain host information.
+ *   if an array was provided for `domain`: an array of domains that are hosted by a green web host if `options.verbose` is false,
+ *     otherwise a dictionary of domain to host information.
  */
 
-function check(domain, db, userAgentIdentifier) {
+function check(domain, optionsOrDb, optionsOrAgentId) {
```
Comment (fershad) [inline src/hosting-node.js:65]: @sfishel18 to move towards consistency between the node and api versions of the hosting check code, let's make _only_ the 2nd parameter the options object.

That is to say, the 2nd parameter could be a string (path to a database file) or an object (options). The 3rd parameter can remain as is (`userAgentIdentifier`). In our docs, we'll guide everyone to use the options object as the recommended way for passing in both the database and userAgentIndentifier. In that way, we can slowly deprecate the third option and move towards simplifying the code across both api and node versions. 

I'll make these changes, but capturing a comment to make my thoughts transparent.

Comment (sfishel18) [reply src/hosting-node.js]: @fershad good call, makes sense to me!

File: src/hosting-node.js [hunk 2]
```diff
@@ -51,64 +52,88 @@ async function getBody(url, userAgentIdentifier) {
 /**
  * Check if a domain is hosted by a green web host.
  * @param {string|array} domain - The domain to check, or an array of domains to be checked.
- * @param {object} db - Optional. A database object to use for lookups.
- * @param {string} userAgentIdentifier - Optional. The app, site, or organisation that is making the request.
- * @returns {boolean|array} - A boolean if a string was provided, or an array of booleans if an array of domains was provided.
+ * @param {string[] | DomainCheckOptions} optionsOrDb - Optional. An object of domain check options, or a database list to use for lookups.
+ * @param {string | DomainCheckOptions} userAgentIdentifier - Optional. The app, site, or organisation that is making the request.
```
Comment (sfishel18) [inline src/hosting-node.js:56]: minor: now that this param can only be the `userAgentIdentifier`, change the type back to `{string}`

**Pull request merged**